src/inline_array/mod.rs

```diff
@@ -88,6 +89,10 @@ pub struct InlineArray<A, T> {
     // 3 start at the beginning of the struct and that all of them are aligned to their maximum
     // alignment.
     //
+    // Unfortunately, we can't use `[A; 0]` to align to actual alignment of the type A, because
+    // it prevents use of InlineArray in recursive types.
+    // We rely on alignment of usize or T to be sufficient, and panic otherwise.
```

Comment:
Just to check, if I create `im::Vector<BigAlign>`, will that panic on me, or will it just figure out to store things out of line? Sorry, I don't have these details memorized any more.

Is panicking a different semver-breaking change, or is it fine this time? Or should we just declare a new version?

Reply:
It panics. This struct never had out-of-line storage, so I didn't want to introduce a completely new behavior. Large alignments were unsound with 0.6.2, so this isn't breaking anything that wasn't already broken in 0.6.2.

Reply:
Hmm, probably. It's better to panic than to UB. But I think long-term, we want `im::Vector<BigAlign>` to Just Work. Panicking in the push (and anything else that adds elements) might make the `Vector` work, if it checks the capacity, though.